osu.Game/Rulesets/Scoring/Legacy/LegacyScoreParser.cs

```diff
@@ -25,6 +25,10 @@ public LegacyScoreParser(RulesetStore rulesets, BeatmapManager beatmaps)
             this.beatmaps = beatmaps;
         }
 
+        protected LegacyScoreParser()
```

Comment:
not sure how safe this is. even though this is a legacy class, may be worth making a base abstract class to avoid confusion?

right now it's not enforced that a consumer must override `GetRuleset`/`GetBeatmap` if they derive the class.